### DISCUSSION TITLE
What additional features do you need?

### DISCUSSION THREAD

**Discussion by cdcsgit** [cdcsgit/lognote#14] [General]
It seems to have all the features I needed.
What other features do you need?

**Comment by procks** [comment]
I have a few suggestions to consider.
I use Lognote mainly for offline browsing.

1. Combine rows with same time and tag like Android Studio does and treat them as one row so I can filter everything with one filter.

![Android Studio](https://github.com/user-attachments/assets/23b7c1b1-6d3f-4229-9d95-03c2040b01c5)
![Lognote](https://github.com/user-attachments/assets/e20fb73b-8f0e-40ce-9887-2819285625af)

2. Ability to create filters for tags.

3. Same filters for full log

4. Create quick filters from dropdown - "Create filter for this message" and tags

5. Vertical scrollable filter list (maybe with checkboxs) so I can use many filters. Also set color for filter.

6. Optional filter name.

7. Sync time in filtered log when clicking on full log.

8. Bookmark not appearing instantly in filtered list

9. Save filtered log with all columns so I can open it again in Lognote

**Reply by cdcsgit** [reply]
Thanks for your suggestions.

Please check the comments for each item.
1. It looks good, but it seems like it needs to be checked to see if it's possible.
2. The filter for tag is already applied. Let me know in more detail.
3. Let me know in more detail.
4. This suggestion seems applicable.
5. Let me know in more detail.
6. The filter name is already applied. Let me know in more detail.
7. Let me know in more detail.
8. Is it means only shows filtered list? (current features available : bookmark + filterd list, bookmark only)
9. It is already applied. Let me know in more detail.

**Comment by jshjs11** [comment]
The filter for log levels will be useful.
For example, I want to see only 'ERROR' and 'WARNING' logs sometimes.

**Reply by cdcsgit** [reply]
If look at the bottom right, there is a combobox to set the level.
Depending on the level you select, it will be displayed in the filter log view(if select "warning", show "warning, error, fatal").

Is your suggestion different from the above?

**Reply by jshjs11** [reply]
I didn't see it was there. That's exactly what I wanted.
Thank you so much!

**Comment by kenjiknk** [comment]
I have a feature suggestion that I believe could make it even better: adding a graphical data visualization feature.
For example, it could allow users to generate:
  - A line chart to track the frequency of a specific error or keyword over time.
  - A pie chart to visualize the proportion of different log levels (Debug, Info, Error, etc.) in a file.

This would make it much easier to spot trends, identify recurring issues, and analyze large log files.
